### ISSUE TITLE
get_order() doesn't return orders already filled

### ISSUE THREAD

**Issue by umitanuki** [alpacahq/pylivetrader#40]
On Alpaca, orders are deleted when fully filled, can no longer do get_order() on their id.
So https://www.quantopian.com/posts/track-orders can't work then.
That's a bad thing because such code can be used for both diagnostics and active things like determining when to change a stop and/or limit (a [partial] fill occurred).
Quite powerful in my view, giving one an edge.
Some folks coming from Quantopian (Q) might be unpleasantly surprised in migrating to Alpaca, using that code or in their own order retrieval purposes.
I myself use that heavily as a core component of my strategy.

The workaround involves extra code less easy to follow, manually saving all order objects with get_order() immediately after every order (a slowdown that risks a missed minute by taking too much time), and rechecking them later to see whether they have disappeared, for the logging of a most recent partial fill, also involving my more extensive diagnostics I have found so useful.

This post has a backtest that demonstrates Q's order objects are always available:
https://www.quantopian.com/posts/error-execution-timeout-1#5bd8be2e7eb72d1bc7e07c1a

The web API can return the full history of orders, but I would say configurable (2 days or so by default) limit (time-based or number-based) would be nice.

**Comment by garyha** [comment]
I'll need order objects available for awhile to implement my strategy on Alpaca.

!["Quantopian backtest"](http://sellerline.com/stocks/quantopian/q2785sm.jpg)

While there are various set commands on Q

```
def initialize(context):
    set_long_only()     # example of actual existing
    set_orders_hold(2)  # conceived just to portray the ballpark of that style
```

... if possible, mere context variable would be comfortable territory instead

```
def initialize(context):
    # conceived, number of days to keep order objects before deletion
    context.days_to_hold_order_objects = 2
```

**Comment by ttt733** [comment]
I'll be filing @garyha 's request for configuring what is stored locally as a separate issue. The `get_order()` functionality has been addressed in v0.0.18. It enables you to query orders without needing to consider how old they are, although it may be slow for some use cases since it's querying the server rather than a local list.

**Comment by garyha** [comment]
Some changes were made and currently using .19 things are much better now.

Link above replaced with https://www.quantopian.com/posts/timing-and-memory-in-this-case-focused-on-the-storing-of-all-orders
